python/lsst/afw/table/match/match.cc

```diff
@@ -35,93 +34,80 @@ using namespace pybind11::literals;
 namespace lsst {
 namespace afw {
 namespace table {
+namespace {
+
+template <typename Record1, typename Record2>
+using PyMatch = py::class_<Match<Record1, Record2>, std::shared_ptr<Match<Record1, Record2>>>;
 
-/// Declare match code templated on two types of catalog
+using PyMatchControl = py::class_<MatchControl, std::shared_ptr<MatchControl>>;
+
+// Declare match code templated on two types of catalog
 template <typename Catalog1, typename Catalog2>
 void declareMatch2(py::module & mod, std::string const & prefix) {
     typedef typename Catalog1::Record Record1;
     typedef typename Catalog2::Record Record2;
-    typedef std::vector<Match<typename Catalog1::Record, typename Catalog2::Record>> MatchList;
-
     using Class = Match<Record1, Record2>;
-    py::class_<Class, std::shared_ptr<Class>> clsMatch(mod, (prefix + "Match").c_str());
-    clsMatch.def(py::init<>());
-    clsMatch.def(py::init<PTR(Record1) const &, PTR(Record2) const &, double>(),
-                 "first"_a, "second"_a, "distance"_a);
+    typedef std::vector<Class> MatchList;
+
+    PyMatch<Record1, Record2> cls(mod, (prefix + "Match").c_str());
+    cls.def(py::init<>());
+    cls.def(py::init<PTR(Record1) const &, PTR(Record2) const &, double>(),
+            "first"_a, "second"_a, "distance"_a);
 
     // struct fields
-    clsMatch.def_readwrite("first", &Match<Record1, Record2>::first);
-    clsMatch.def_readwrite("second", &Match<Record1, Record2>::second);
-    clsMatch.def_readwrite("distance", &Match<Record1, Record2>::distance);
+    cls.def_readwrite("first", &Class::first);
+    cls.def_readwrite("second", &Class::second);
+    cls.def_readwrite("distance", &Class::distance);
 
     // Free Functions
     mod.def("unpackMatches", &unpackMatches<Catalog1, Catalog2>, "matches"_a, "cat1"_a, "cat2"_a);
-
     mod.def("matchRaDec",
             (MatchList (*)(Catalog1 const &, Catalog2 const &, geom::Angle, MatchControl const &))
              matchRaDec<Catalog1, Catalog2>, "cat1"_a, "cat2"_a, "radius"_a, "mc"_a=MatchControl());
-    // The following is deprecated; consider changing the code instead of wrapping it:
-    // mod.def("matchRaDec",
-    //         (MatchList (*)(Catalog1 const &, Catalog2 const &, geom::Angle, bool))
-    //          matchRaDec<Catalog1, Catalog2>, "cat1"_a, "cat2"_a, "radius"_a, "closest"_a);
```

Comment:
Consider leaving this comment in until we can properly  deprecate the method -- judging from how often the old forms were used in the stack, they were quire popular.